src/_utils/focusVisible/FocusVisibleProvider.unit.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useContext } from 'react'
+import { mount } from 'enzyme'
+import cc from 'classcat'
+import FocusVisibleProvider, { FocusVisibleContext } from './FocusVisibleProvider'
+
+const ChildComponent = () => {
+  const { isInitialized, hadKeyboardEvent } = useContext(FocusVisibleContext)
+  return (
+    <button
+      className={cc([
+        {
+          initialized: isInitialized,
+          keyboardEvent: hadKeyboardEvent,
+        },
+      ])}
+    >
+      Test
+    </button>
+  )
+}
+
+describe('FocusVisibleProvider', () => {
+  it('Should provide "true" value for isInitialized and "false" for hadKeyboardEvent', () => {
+    const wrapper = mount(
+      <FocusVisibleProvider>
+        <ChildComponent />
+      </FocusVisibleProvider>,
+    )
+
+    const button = wrapper.find('button')
+
+    expect(button.hasClass('initialized')).toBe(true)
+    expect(button.hasClass('keyboardEvent')).toBe(false)
+  })
```

Comment:
Don't find a way to simulate key events on `document` with `element.simulate('keydown', { key: 'Tab' }`.

Reply:
Tried this idea with no success:
https://bytes.babbel.com/en/articles/2019-01-22-testing-global-event-listener-within-react-component.html

Reply:
I did something similar for the search form look at
https://github.com/blablacar/ui-library/blob/master/src/searchForm/SearchForm.unit.tsx#L83

Reply:
Yes it works in the end.
I just don't find a way to check the provider value on specific tag element.
I mean for example, fire a keyboard event on a Textarea with typing key (Spacebar/Enter)